src/server/req_jobobit.c

```diff
@@ -627,6 +627,9 @@ rel_resc(job *pjob)
 		if (pjob->ji_rerun_preq->rq_conn != PBS_LOCAL_CONNECTION)
 			conn = get_conn(pjob->ji_rerun_preq->rq_conn);
 
+		if (pjob->ji_pmt_preq != NULL)
+			reply_preempt_jobs_request(PBSE_NONE, PREEMPT_METHOD_REQUEUE, pjob);
+
```

Comment:
This can't go in rel_resc().  rel_resc() is a very generic function that is called whenever resources need to be released.  It is not just for requeuing jobs.  If you put it here, then any time rel_resc() is called, the preemption would be replied to thinking it was for requeue.  I think the place you want is just after the rel_resc() call is made in on_job_rerun() at line 1422 in req_jobobit.c.

Reply:
Code change was inside a rerun request check, which run only for requeuing job, but I do feel the better place for replying preemption is from on_job_rerun(), thanks for pointing to the right function. I have made the suggested changes.